packages/stats-py/tests/test_tier3.py

```
@@ -0,0 +1,190 @@
+    communalities = _table(result, "communalities")
+    for row in communalities["rows"]:
+        assert 0.0 <= row[2] <= 1.0 + 1e-9  # extraction
+        assert 0.0 <= row[3] <= 1.0 + 1e-9  # MSA per item
+    variance = _table(result, "variance")
+    eigenvalues = [row[1] for row in variance["rows"]]
+    assert sum(eigenvalues) == pytest.approx(len(ITEMS_8), abs=0.01)  # trace = p (tabel dibulatkan 3 desimal)
+    assert eigenvalues == sorted(eigenvalues, reverse=True)
+    matrix = _table(result, "component_matrix")
+    assert len(matrix["columns"]) == 2 + k
+    # Communality = jumlah kuadrat loading baris (invarian rotasi).
+    for c_row, m_row in zip(communalities["rows"], matrix["rows"]):
```

Ruff (0.15.20)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)